draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -209,6 +209,81 @@ WebRTC applications running in a third-party browser execution context (i.e., a
 context that has a different origin than the top-level execution context), or a
 private browser execution context.
 
+Security Considerations {#security}
+======================
+
+mDNS Message Flooding
+---------------------
+
+The implementation of this proposal requires the mDNS querying capability of the
+browser for registering mDNS names or adding remote ICE host candidates with
+such names. It also requires the mDNS responding capability of either the
+browser or the operating platform of the browser for registering, removing or
+resolving mDNS names. In particular,
+
+- the registration of name requires optional probing queries and mandatory
+announcing responses ({{RFC6762}, Section 8}), and this is performed at the
+beginning of ICE gathering;
+
+- the addition of remote ICE host candidates with mDNS names generates mDNS
+queries for names of each candidate;
+
+- the removal of names could happen when the execution context of the ICE agent
+is destroyed in an implementation, and goodbye responses should be sent to
+invalidate records generated by the ICE agent in the local network
+({{RFC6762}}, Section 10.1).
+
+A malicious Web application could flood the local network with mDNS messages by:
+
+- creating execution contexts that create ICE agents and start gathering of
+  local ICE host candidates;
+
+- destroying these local candidates soon after the name registration is done;
+
+- adding fictitious remote ICE host candidates with mDNS names.
+
+
+{{RFC6762}} defines a per-record multicast rate limiting rule, in which a given
+record on a given interface cannot be sent less than one second since its last
+transmission. This rate limiting rule however does not mitigate the above
+attacks, in which new names, hence new records, are constantly created and sent.
+A per-message multicast rate limiting rule on a given interface MUST be provided
```

Comment:
We already have a browser-wide rate limit for ICE checks, so I would recommend we make this browser-wide as well. It will only matter in pathological cases.

```suggestion
A browser-wide mDNS message rate limit MUST be provided for all messages
that can be indirectly dispatched by a web application, namely the probing
queries, announcement responses, resolution queries, and goodbye responses
associated with mDNS.
```